File: C#/TicketBooker/TicketBooker/Tickets.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace TicketBooker
{
    public class Tickets
    {
        public List<Ticket> TicketList = new List<Ticket>();
        public int TicketPrice;

        public string GetReport()
        {
            var standardTicketTotal = 0.0;
            var studentTicketTotal = 0.0;
            foreach (var tickets in TicketList)
            {
                switch (tickets.CustomerType)
                {
                    case Ticket.CustomerTypes.Standard:
                        standardTicketTotal += TicketPrice;
                        break;
                    case Ticket.CustomerTypes.Student:
                        studentTicketTotal += TicketPrice * 0.6;
                        break;
                }
            }

            var total = standardTicketTotal + studentTicketTotal;
            var reportOutput = "** SALES REPORT **" + Environment.NewLine + Environment.NewLine;
            reportOutput += "Number of Tickets Sold" + Environment.NewLine;
            var standardCount = 0;
            foreach (var ticket in TicketList)
                if (ticket.CustomerType == Ticket.CustomerTypes.Standard)
                    standardCount++;
            reportOutput += "  Standard: " + standardCount + Environment.NewLine;
            var studentCount = 0;
            foreach (var ticket in TicketList)
                if (ticket.CustomerType == Ticket.CustomerTypes.Student)
                    studentCount++;
            reportOutput += "  Student: " + studentCount + Environment.NewLine;
            reportOutput += Environment.NewLine;
            reportOutput += "Income" + Environment.NewLine;
            reportOutput += "  Standard Total = £" + standardTicketTotal + Environment.NewLine;
            reportOutput += "  Student Total = £" + studentTicketTotal + Environment.NewLine;
            reportOutput += "  Total = £" + total;
            return reportOutput;
        }

        public void Add(Ticket ticket)
        {
            TicketList.Add(ticket);
        }
    }
}

File: Java2/src/main/java/ticketbooker/Tickets.java
package ticketbooker;

import java.util.ArrayList;

public class Tickets {
    private final String newLine = System.getProperty("line.separator");
    public ArrayList<Ticket> ticketList = new ArrayList<Ticket>();
    public int ticketPrice;

    public String getReport() {
        Double standardTicketTotal = 0.0;
        Double studentTicketTotal = 0.0;
        for (Ticket tickets : ticketList) {
            switch (tickets.CustomerType) {
                case Standard:
                    standardTicketTotal += ticketPrice;
                    break;
                case Student:
                    studentTicketTotal += ticketPrice * 0.6;
                    break;
            }
        }
        Double total = standardTicketTotal + studentTicketTotal;
        String reportOutput = "** SALES REPORT **" + newLine + newLine;
        reportOutput += "Number of Tickets Sold" + newLine;
        long standardTicketCount = ticketList.stream().filter(t -> t.CustomerType == Ticket.CustomerTypes.Standard).count();
        reportOutput += "  Standard: " + standardTicketCount + newLine;
        long studentTicketCount = ticketList.stream().filter(t -> t.CustomerType == Ticket.CustomerTypes.Student).count();
        reportOutput += "  Student: " + studentTicketCount + newLine;
        reportOutput += newLine;
        reportOutput += "Income" + newLine;
        reportOutput += "  Standard Total = £" + standardTicketTotal + newLine;
        reportOutput += "  Student Total = £" + studentTicketTotal + newLine;
        reportOutput += "  Total = £" + total;
        return reportOutput;
    }

    public void add(Ticket ticket) {
        ticketList.add(ticket);
    }
}